src/main/java/uk/gov/companieshouse/api/strikeoffobjections/processor/ObjectionProcessor.java

```diff
@@ -0,0 +1,90 @@
+import uk.gov.companieshouse.api.strikeoffobjections.common.ApiLogger;
+import uk.gov.companieshouse.api.strikeoffobjections.common.LogConstants;
+import uk.gov.companieshouse.api.strikeoffobjections.exception.InvalidObjectionStatusException;
+import uk.gov.companieshouse.api.strikeoffobjections.exception.ObjectionNotFoundException;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.Objection;
+import uk.gov.companieshouse.api.strikeoffobjections.model.entity.ObjectionStatus;
+import uk.gov.companieshouse.api.strikeoffobjections.service.IObjectionService;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Processes an Objection
+ *
+ * Will only process Objection if status = SUBMITTED
+ * Calls Chips to place stop against company
+ * Sends email
+ */
+@Component
+public class ObjectionProcessor {
+
+    private static final String INVALID_START_STATUS_MSG =
+            "Objection %s has status %s. Cannot process unless status = SUBMITTED";
+    private static final String LOG_OBJECTION_ID_KEY = LogConstants.OBJECTION_ID.getValue();
+
+    private IObjectionService objectionService;
+    private ApiLogger apiLogger;
+
+    @Autowired
+    public ObjectionProcessor(IObjectionService objectionService,
+                              ApiLogger apiLogger) {
+        this.objectionService = objectionService;
+        this.apiLogger = apiLogger;
+    }
+
+    /**
+     * Process the specified Objection
+     * Only processes if status is SUBMITTED
+     *
+     * @param httpRequestId http request id used for logging
+     * @param objectionId   id of the objection to process
+     * @throws ObjectionNotFoundException      if objectionId not found in database
+     * @throws InvalidObjectionStatusException if the Objection is not currently in status SUBMITTED when this is called
+     */
+    public void process(String httpRequestId, String objectionId)
+            throws ObjectionNotFoundException, InvalidObjectionStatusException {
+
+        Map<String, Object> logMap = new HashMap<>();
+        logMap.put(LOG_OBJECTION_ID_KEY, objectionId);
+        apiLogger.infoContext(httpRequestId, "Starting objection processing", logMap);
+
+        Objection objection = getObjectionForProcessing(httpRequestId, objectionId);
+
+        // TODO update status to processing
+
+        // TODO OBJ-139/OBJ-20 do chips sending
+
+        // TODO update status to chips sent
+
+        // TODO OBJ-113 do email sending
+
+        // TODO update status to processed
+
+    }
+
+    private Objection getObjectionForProcessing(String httpRequestId, String objectionId)
+            throws ObjectionNotFoundException, InvalidObjectionStatusException {
+
+        // get objection
+        Objection objection = objectionService.getObjection(httpRequestId, objectionId);
+
+        // if status not = submitted, throw exception
+        if (objection != null && ObjectionStatus.SUBMITTED != objection.getStatus()) {
```

Comment:
I wonder if this check should be more specific and only allow processing when status is 'OPEN'. This would protect against 'PROCESSED' objections potentially being processed again, when we come to set that status...

Reply:
it should be ok, if status != submitted it will throw an InvalidObjectionStatusException and won't process it. My comment may have caused confusion, it used "not =" instead of "!="

Reply:
I guess this is ok, it's essentially repeating the logic check that's done in the controller PATCH method so in theory it will never fail/trigger, but no harm in repeating it.

The one thing that both the controller and this processor will allow though (I think) is for all the processing to be performed multiple times if the controller PATCH method is invoked multiple times with a status of 'SUBMITTED' - we'd further checks in the service method probably to implement that state machine your talking about and not allow invalid transitions via calls to the PATCH method. I actually think this is quite important with regard to robustness of the service. What do you think?

Reply:
As discussed, created story OBJ-177